make/hotspot/lib/CompileJvm.gmk

```diff
@@ -83,26 +83,22 @@ CFLAGS_VM_VERSION := \
 ################################################################################
 # Disabled warnings
 
-DISABLED_WARNINGS_gcc := parentheses comment unknown-pragmas address \
-    delete-non-virtual-dtor char-subscripts array-bounds int-in-bool-context \
-    ignored-qualifiers  missing-field-initializers implicit-fallthrough \
-    empty-body strict-overflow sequence-point maybe-uninitialized \
-    misleading-indentation cast-function-type shift-negative-value
-
-ifeq ($(call check-jvm-feature, zero), true)
-  DISABLED_WARNINGS_gcc += return-type switch clobbered
-endif
+DISABLED_WARNINGS_gcc := array-bounds comment delete-non-virtual-dtor \
+    empty-body ignored-qualifiers implicit-fallthrough int-in-bool-context \
+    maybe-uninitialized missing-field-initializers parentheses \
+    shift-negative-value unknown-pragmas
+
+DISABLED_WARNINGS_clang := ignored-qualifiers sometimes-uninitialized \
+    missing-braces delete-non-abstract-non-virtual-dtor unknown-pragmas
```

Comment:
Shouldn't shift-negative-value be in the clang list too?

Reply:
Well, there is currently no instance of clang complaining about this. This could be due to:

* This warning does not really exist on clang
* Or it is not enabled by our current clang flags
* Or the code which triggers the warning in gcc is not reached by clang
* Or clang is smarter than gcc and can determine that the usage is ok after all
* Or clang is dumber than gcc and does not even see that there could have been a problem...

...

I'm kind of reluctant to add warnings to this list that have not occurred for real. My suggestion is that we add it here if we ever see it making incorrect claims. Ok?

Reply:
I'm worried about someone encountering it and uglifying code to work around
it. OTOH, I don't know why we're not seeing this warning with clang, as there
is shared code that should trigger it (and does for gcc). So I'm okay with it
as is.